ocaml/lambda/matching.ml

```diff
@@ -3811,13 +3818,16 @@ let for_tupled_function ~scopes loc kind paraml pats_act_list partial =
 
 let flatten_pattern size p =
   match p.pat_desc with
-  | Tpat_tuple args -> args
+  | Tpat_tuple args ->
+    List.map
+      (fun (lbl, p) -> if Option.is_some lbl then raise Cannot_flatten else p)
+      args
```

Comment:
Would `List.map snd args` be wrong here?

If you look at how this is used in `transl_tupled_function`, I think raising `Cannot_flatten` for labeled tuples means we get worse compilation for a function with a labeled tuple as an argument compared to the equivalent unlabeled tuple.  To see this, try `ocamlopt -dlambda` on:

```ocaml
let f (x, y) = x + y

let f (~~(~x; ~y)) = x + y
```

Reply:
This is right; `List.map snd` is better. Thanks!

Just for reference, the dlambda output changes from
```
(seq
  (let
    (f/267 =
       (function (param/277[int], param/278[int]) : int
         (+ param/277 param/278)))
    (setfield_ptr(root-init) 1 (global Dlambda!) f/267))
  (let
    (f/272 =
       (function {nlocal = 0}
         param/275[(consts ()) (non_consts ([0: [int], [int]]))] : int
         (+ (field 0 param/275) (field 1 param/275))))
    (setfield_ptr(root-init) 0 (global Dlambda!) f/272))
  0)
```
to
```
(seq
  (let
    (f/267 =
       (function (param/277[int], param/278[int]) : int
         (+ param/277 param/278)))
    (setfield_ptr(root-init) 1 (global Dlambda!) f/267))
  (let
    (f/272 =
       (function (param/279[int], param/280[int]) : int
         (+ param/279 param/280)))
    (setfield_ptr(root-init) 0 (global Dlambda!) f/272))
  0)
```